backend/de.metas.util/src/main/java/de/metas/util/time/DurationUtils.java

```diff
@@ -35,10 +38,32 @@
 @UtilityClass
 public class DurationUtils
 {
-	public static Duration toDuration(@NonNull final BigDecimal durationBD, @NonNull final TemporalUnit unit)
+	private static final List<TemporalUnit> supportedTemporalUnits = ImmutableList.of(ChronoUnit.YEARS, ChronoUnit.MONTHS, ChronoUnit.DAYS, ChronoUnit.HOURS, ChronoUnit.MINUTES, ChronoUnit.SECONDS, ChronoUnit.NANOS);
+
+	public static Duration toWorkDuration(final @NonNull BigDecimal durationBD, final @NonNull TemporalUnit unit)
+	{
+		BigDecimal duration = durationBD;
+		TemporalUnit currentUnit = unit;
+		long durationLong = -1;
+		while (durationLong < 0)
+		{
+			try
+			{
+				durationLong = duration.longValueExact();
+			}
+			catch (final ArithmeticException ae)
```

Comment:
avoid using exceptions to control the flow.

pls check if `NumberUtils.stripTrailingDecimalZeros(duration).scale() == 0` might do the job.

Reply:
..or `scale() <= 0`

Reply:
makes sense, changed.